libcloud/compute/drivers/dimensiondata.py

```diff
@@ -815,6 +816,24 @@ def ex_get_location_by_id(self, id):
                 filter(lambda x: x.id == id, self.list_locations()))[0]
         return location
 
+    def ex_wait_for_state(self, state, func, *args, **kwargs):
+        """
+        Wait for the function which returns a instance
+        with field status to match
+
+        Keep polling func until one of the desired states is matched
+
+        :param state: Either the desired state (`str`) or a `list` of states
+        :type  state: ``str`` or ``list``
+
+        :param  func: The function to call, e.g. ex_get_vlan
+        :type   func: ``function``
+
+        :param  kwargs: The arguments for func
+        :type   kwargs: Keyword arguments
+        """
+        self.connection.wait_for_state(state, func, *args, **kwargs)
```

Comment:
Same here - if you update the original function to return the result you can also update this one.